### ISSUE TITLE
"availability" should be a boolean, rename to "discontinued"

### ISSUE THREAD

**Issue by broofa** [JohnCoker/thrustcurve3#35]
**tl;dr: 'availability' should be renamed (and redefined) to `discontinued`.**

I was going to add this as a comment to #34 but it kind of took on a life of its own.  Apologies in advance for the wall of text.

I've been finding the "availability" thing curiously difficult to work with.  The meaning of "availability" is obvious - "Is motor currently available from a commercial manufacturer?"_  - but in practice it's proving to be overly complicated.

For starters, there's the namespace of possible values. The DB [allows for](https://github.com/JohnCoker/thrustcurve3/blob/0a0138609eed70ffec4726301bf856b2e01e0fd0/database/schema/schema.js#L28) availability values of "regular", "OOP", and "occasional", but the API docs (specifically `SearchRequest`) also allow for "all" and "available".

So five possible values.  But are they all necessary? IMHO, no.

The "all" option is pretty easy to dismiss.  It's wholly redundant with omitting the query parameter altogether. (And, note, that this is the only param that works this way.  The other params require you to omit the value.)
 
The "OOP" and "occasional" params might be interesting if they actually worked, but [they don't](https://www.thrustcurve.org/api/v1/search.json?availability=OOP).  The only DB value you can actually query for is "regular".  But doing so gets you almost the same results as querying for "available".  Almost.

There's those two pesky motors I mentioned in #34, the Loki N3800 and Estes D11, that are "available" but not "regular".  So what's going on there.   It took me a while, but I eventually figured out that these are the only two motors marked as "occasional".  Which begs the question, "What does 'occasional' mean???"  After all, these motors couldn't be more different.  A "D" and an "N" motor.  One from Estes, one from Loki ("who???").

After some googling, I think I've figured out the reason for the D11 being "occasional".   I found [this post](https://www.rocketryforum.com/threads/who-wants-estes-d11-p-motors.59787/) that says the D11 is produced on demand by Estes.  But... that post is 8 years old.  Is that still the case?  There really needs to be more context here.  And the N3800?  'Couldn't find anything.  So, again, definitely needs more context.

Without that additional context, the "occasional" state is effectively meaningless.  "Can we or can't we get these motors?!?"

Thus, I would argue the following ...
- "all" should go away (redundant with no query param)
- "occasional" should go away (not useful w/out more context, doesn't apply to enough motors to warrant inclusion)
- "available" should go away (redundant with "regular" once you remove "occasional")
- ... leaving you with just "regular" and "OOP"

So two meaningful "availability" states... i.e. a boolean.

And if this is going to be a boolean property, can it be defined in a way that makes the meaning completely unambiguous?   I think so.  If we rename this to "`discontinued`", it is immediately clear what the meaning is:  It means the motor is no longer manufactured or sold.  Period.

I'm curious what your take is on this.  I fully understand if it's, "leaving as is for legacy reasons, will consider for v2".  But if this makes sense maybe maybe the fix for #34 should be to add a `discontinued` property as described here, rather than `availability`, as that would still be backward compatible and avoid propagating the "availability" complexity problems.

**Comment by broofa** [comment]
fwiw, I've gone ahead and switched the thrustcurve-db data over to having a 'discontinued' property and published as v0.3.4.  This picks up the changes you just made to the D11 and N3800, too.

**Comment by broofa** [comment]
Followup: with the `availability` states now being consistent in the DB (either 'OOP' or 'regular' for all motors), I'm removing the `discontinued` field from thrustcurve-db.  The closer we can keep the datastructures between these two systems, the better.
